spec/unit/facter/systemd_spec.rb

```diff
@@ -12,7 +12,7 @@
       end
       let(:facts) { {:kernel => :linux} }
       it do
-        Facter::Util::Resolution.expects(:exec).with('ps -p 1 -o comm=').returns('systemd')
```

Comment:
I confess that I've got a lot to learn about rspec.

I like your idea of testing for /run/systemd as a lighter-weight solution, though.